### PR TITLE
Add PyTorch & GPU support for DNSMOS calculation

### PR DESCRIPTION
**Pull request by Emrys365**
## What?

This PR adds support for PyTorch-based DNSMOS calculation in addition to the existing ONNX-based one. This is based on a third-party package [`onnx2torch`](https://github.com/ENOT-AutoDL/onnx2torch) which converts the ONNX-format model to a PyTorch module.

## Why?

Current DNSMOS calculation basically relies on the ONNX wrapper, where the GPU-based processing is not well supported. By converting the ONNX-format model to a PyTorch module using `onnx2torch`, we can now easily use CUDA devices to greatly speed up the calculation.

In addition, since the model is converted to a PyTorch module, gradient backpropagation is also supported, which can be used to provide supervision during training later on.

### CODE REVIEW

**Comment by sw005320** [comment]
It requires additional packages for the test https://github.com/espnet/espnet/actions/runs/6872983326/job/18692312021?pr=5548#step:9:2429

**Comment by sw005320** [comment]
How about only adding it in https://github.com/espnet/espnet/blob/master/ci/install.sh?

**Comment by Emrys365** [comment]
OK. I'll try it.

**Comment by sw005320** [comment]
Thanks a lot!
